knowledge-base/adr/0020-generator-options.md

```diff
@@ -0,0 +1,60 @@
+   Flatten the directory to be `<outputDir>`.
+   Throw an error if input is a directory (with multiple files).
+4. Keep it as is. (decided)
+5. Rename the client:
+   1. `openapi-generator`
+   2. `sap-cloud-sdk-openapi-generator`
+
+## Consequences 1
+
+The given `outputDir` is the actual output directory.
+Depending on the decided option, the behavior might be considered inconsistent by default.
+
+## Context 2
+
+The option naming is slighly inconsistent and quite long at times.
+Some of the defaults might lead to many files being generated and transpiled by default - potentially more than necessary.
+We should discuss the details of those defaults and make a concious decision on whether to keep or adjust them.
+
+## Decision 2
+
+| Current Name, Aliases                                    | Future Name, Aliases             | Current Behavior                                                                                  | Future Behavior                                                                                                                                              | Allow-no |
+| :------------------------------------------------------- | :------------------------------- | :------------------------------------------------------------------------------------------------ | :----------------------------------------------------------------------------------------------------------------------------------------------------------- | :------: |
+| `--inputDir`, `-i` _(required)_                          | `--input`, `-i` _(required)_     | Input directory/file                                                                              | _same_                                                                                                                                                       |  false   |
+| `--outputDir`, `-o` _(required)_                         | `--outputDir`, `-o` _(required)_ | Output directory                                                                                  | _same_                                                                                                                                                       |  false   |
+| `--clearOutputDir`                                       | `--clearOutputDir`               | Delete all files in output directory                                                              | _same_                                                                                                                                                       |  false   |
+| `--generateJs`                                           | `--transpile`, `-t`              | Transpiles, default true                                                                          | Transpiles, default false. Can only be set if tsconfig is enabled. This should be explicitly stated in the documentation.                                    |  false   |
```

Comment:
Please correct me.
I thought, we agreed that, when `-t` was true, then the default `tsconfig` would be used, unless user provided a custom `tsconfig`. 
Without a default `tsconfig`, there would be additional effort to transpile the generated client for a js project.

Reply:
Good point, I thought it was already up to date. Done.